server/auth/types/saml/routes.ts

```diff
@@ -131,15 +131,17 @@ export class SamlAuthRoutes {
         }
 
         try {
-          const credentials = await this.securityClient.authToken(
+          const credentials = await this.securityClient.authToken({
             requestId,
-            request.body.SAMLResponse,
-            undefined
-          );
+            samlResponse: request.body.SAMLResponse,
+            acsEndpoint: undefined,
+            authRequestType: AuthType.SAML,
```

Comment:
I have used to set it explicitly to avoid redirecting auto-login as anonymous upon login failure. This was the issue I faced with basic auth when passing wrong credentials led to auto login as anonymous because backend couldn't differentiate anonymous from non-anonymous logon requests since front-end didn't specify it in particular.

Reply:
I think its possible to pass only where necessary: https://github.com/cwperks/security-dashboards-plugin/pull/6

There's a separate issue on the log out flow for multi-auth where the logic for automatically redirect to anonymous for multi-auth is unnecessary: https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/multiple/multi_auth.ts#L169-L176

I think its unnecessary because when multi-auth is enabled, login with anonymous should always be initiated by pushing the button `Login as Anonymous`. From what I understand, this block is necessary in the [basic_auth handler](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/basic/basic_auth.ts#L114-L121) because security-dashboards-plugin needs to support the case where anonymous login is the _only_ form of login.

Reply:
I tried this, and without passing it explicitly here, the login threw 500 error when i logged out and tried to log in again.

Reply:
I've removed all changes out of scope of this bug fix to keep the review much simpler.